features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -235,9 +244,45 @@ public Response getDeviceConfigsByInterface(Integer ipInterfaceId, String config
 
     /** {@inheritDoc} */
     @Override
-    public void deleteDeviceConfig(long id) {
-        throw new UnsupportedOperationException("Delete device config not yet supported.");
-        // deviceConfigDao.delete(id);
+    public Response deleteDeviceConfig(String id) {
+        if (Strings.isNullOrEmpty(id)) {
+            LOG.debug("deleteDeviceConfig: empty or null id supplied by request");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+
+        final String idParamPattern = "\\d+(, ?\\d+)*";
+        var pattern = Pattern.compile(idParamPattern);
+        var matcher = pattern.matcher(id);
+
+        if (!matcher.matches()) {
+            LOG.debug("deleteDeviceConfig: invalid id param supplied by request");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+
+        List<Long> ids = Arrays.stream(id.split(","))
+                .filter(s -> !Strings.isNullOrEmpty(s))
```

Comment:
The pattern allows spaces after the comma. This needs handling here.

Reply:
trim() method added to remove extra space

Reply:
reg expression changed to take care of spaces